### PR TITLE
Use new EXEC code for classic protocol, remove old.

### PR DESCRIPTION
**Pull request by jimis**

### CODE REVIEW

**Comment by jimis** [comment]
trigger build

**Comment by jimis** [comment]
While Jenkins is going on, I'm manually testing with 3.6.x cf-runagent, just to make sure that the new cf-serverd code is compatible with the old client.

File: cf-serverd/server_classic.c [hunk 1]
```diff
@@ -1142,36 +1142,28 @@ int BusyWithClassicConnection(EvalContext *ctx, ServerConnectionState *conn)
...
 
-        if (!MatchClasses(ctx, conn))
-        {
-            Log(LOG_LEVEL_INFO,
-                "REFUSAL due to failed class/context match");
-            Terminate(conn->conn_info);
-            return false;
-        }
+        bool b = DoExec2(ctx, conn, args,
+                         sendbuffer, sizeof(sendbuffer));
```
Comment (kacfengine) [inline cf-serverd/server_classic.c]: Should this be called `DoExec` now? :-)

**Comment by kacfengine** [comment]
+1 for code removal! Looks good!

File: cf-serverd/server_classic.c (hunk 1 above)
Comment (jimis) [inline cf-serverd/server_classic.c]: Indeed. I think I'll leave it like that to honour the legacy, and hopefully we won't ever see a "3" version... ;-)

**Comment by jimis** [comment]
Manual tests OK, auto tests in Jenkins mostly green, so I'll merge as soon as all platforms are finished. Thanks for reviewing!

**Pull request merged**